From 227ccd784c294e12032afdcaca496a7b1a41fc36 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 30 Sep 2020 01:17:15 -0400
Subject: [PATCH 006/148] Revert "SystemUI: Move charging animation above FOD
 icon if In-display fingerprint is active"

This reverts commit 3bcc13533a76709a28467867e91c8255fffd97ea.
---
 .../KeyguardIndicationController.java         | 58 -------------------
 1 file changed, 58 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/KeyguardIndicationController.java b/packages/SystemUI/src/com/android/systemui/statusbar/KeyguardIndicationController.java
index 0b78f943611..f0164df175a 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/KeyguardIndicationController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/KeyguardIndicationController.java
@@ -21,11 +21,9 @@ import android.animation.AnimatorListenerAdapter;
 import android.animation.ValueAnimator;
 import android.app.admin.DevicePolicyManager;
 import android.content.Context;
-import android.content.pm.PackageManager;
 import android.content.res.ColorStateList;
 import android.content.res.Resources;
 import android.graphics.Color;
-import android.graphics.Point;
 import android.hardware.biometrics.BiometricSourceType;
 import android.hardware.face.FaceManager;
 import android.hardware.fingerprint.FingerprintManager;
@@ -41,10 +39,8 @@ import android.provider.Settings;
 import android.text.TextUtils;
 import android.text.format.Formatter;
 import android.util.Log;
-import android.view.Display;
 import android.view.View;
 import android.view.ViewGroup;
-import android.view.WindowManager;
 
 import com.airbnb.lottie.LottieAnimationView;
 
@@ -79,9 +75,6 @@ import java.io.PrintWriter;
 import java.text.NumberFormat;
 import java.util.IllegalFormatConversionException;
 
-import lineageos.app.LineageContextConstants;
-import vendor.lineage.biometrics.fingerprint.inscreen.V1_0.IFingerprintInscreen;
-
 /**
  * Controls the indications and error messages shown on the Keyguard
  */
@@ -112,7 +105,6 @@ public class KeyguardIndicationController implements StateListener,
     private KeyguardIndicationTextView mDisclosure;
     private LottieAnimationView mChargingIndicationView;
     private int mChargingIndication = 1;
-    private int mFODPositionY = 0;
     private final UserManager mUserManager;
     private final IBatteryStats mBatteryInfo;
     private final SettableWakeLock mWakeLock;
@@ -250,17 +242,6 @@ public class KeyguardIndicationController implements StateListener,
         mChargingIndicationView = (LottieAnimationView) indicationArea.findViewById(
                 R.id.charging_indication);
         updateChargingIndicationStyle();
-        if (hasActiveInDisplayFp()) {
-            try {
-                IFingerprintInscreen daemon = IFingerprintInscreen.getService();
-                mFODPositionY = daemon.getPositionY();
-            } catch (RemoteException e) {
-                // do nothing
-            }
-            if (mFODPositionY <= 0) {
-                mFODPositionY = 0;
-            }
-        }
         updateIndication(false /* animate */);
     }
 
@@ -572,37 +553,6 @@ public class KeyguardIndicationController implements StateListener,
 
     private void updateChargingIndication() {
         if (mChargingIndication > 0 && !mDozing && mPowerPluggedIn) {
-            if (hasActiveInDisplayFp()) {
-                if (mFODPositionY != 0) {
-                    // Get screen height
-                    WindowManager windowManager = mContext.getSystemService(WindowManager.class);
-                    Display defaultDisplay = windowManager.getDefaultDisplay();
-                    Point size = new Point();
-                    defaultDisplay.getRealSize(size);
-                    int screenHeight = size.y;
-                    // Correct FOD position if cutout is hidden
-                    int statusbarHeight = mContext.getResources().getDimensionPixelSize(
-                            com.android.internal.R.dimen.status_bar_height_portrait);
-                    boolean cutoutMasked = mContext.getResources().getBoolean(
-                            com.android.internal.R.bool.config_maskMainBuiltInDisplayCutout);
-                    int fodPositionY = mFODPositionY;
-                    if (cutoutMasked) {
-                        fodPositionY = mFODPositionY - statusbarHeight;
-                    }
-                    // Get indication text height
-                    int textViewHeight = mTextView.getMeasuredHeight();
-                    // Get bottom margin height
-                    int marginBottom = mContext.getResources().getDimensionPixelSize(
-                            R.dimen.keyguard_indication_margin_bottom_fingerprint_in_display);
-                    // Calculate charging indication margin
-                    int animationMargin = (screenHeight - fodPositionY) - (textViewHeight + marginBottom) + 10;
-                    // Set position of charging indication
-                    ViewGroup.MarginLayoutParams params =
-                            (ViewGroup.MarginLayoutParams) mChargingIndicationView.getLayoutParams();
-                    params.setMargins(0, 0, 0, animationMargin);
-                    mChargingIndicationView.setLayoutParams(params);
-                }
-            }
             mChargingIndicationView.setVisibility(View.VISIBLE);
             mChargingIndicationView.playAnimation();
         } else {
@@ -610,14 +560,6 @@ public class KeyguardIndicationController implements StateListener,
         }
     }
 
-    private boolean hasActiveInDisplayFp() {
-        PackageManager packageManager = mContext.getPackageManager();
-        boolean hasInDisplayFingerprint = packageManager.hasSystemFeature(LineageContextConstants.Features.FOD);
-        int userId = KeyguardUpdateMonitor.getCurrentUser();
-        FingerprintManager fpm = (FingerprintManager) mContext.getSystemService(Context.FINGERPRINT_SERVICE);
-        return hasInDisplayFingerprint && fpm.getEnrolledFingerprints(userId).size() > 0;
-    }
-
     // animates textView - textView moves up and bounces down
     private void animateText(KeyguardIndicationTextView textView, String indication) {
         int yTranslation = mContext.getResources().getInteger(
-- 
2.17.1

